apps/prairielearn/src/migrations/20241007203121_draft_questions.sql

```diff
@@ -0,0 +1,35 @@
+ALTER TABLE questions
+ADD COLUMN IF NOT EXISTS is_draft boolean NOT NULL DEFAULT false;
+
+ALTER TABLE pl_courses
+ADD COLUMN IF NOT EXISTS draft_number integer NOT NULL DEFAULT 0;
+
+CREATE TABLE IF NOT EXISTS draft_question_metadata (
+  id bigserial NOT NULL PRIMARY KEY,
+  question_id BIGINT REFERENCES questions (id) ON DELETE SET NULL ON UPDATE CASCADE,
+  created_at TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  created_by BIGINT REFERENCES users (user_id) ON DELETE SET NULL ON UPDATE CASCADE,
+  updated_by BIGINT REFERENCES users (user_id) ON DELETE SET NULL ON UPDATE CASCADE
+);
+
+CREATE TYPE ai_prompt_type AS ENUM(
+  'initial_prompt',
+  'human_revision',
+  'autorevision'
```

Comment:
Nits, because this would probably bug me going forwards:

- Can we use `initial` instead of `initial_prompt`? "prompt" is already baked into the enum name.
- Can we use `auto_revision` instead of `autorevision`? If I were to write that in a sentence, I'd probably write "auto-revision" instead of "autorevision".

Reply:
changed in ddcadb9e5d153fe8dd8517715ea3b9fe9154369e.